src/catalog/src/information_schema.rs

```diff
@@ -51,42 +56,93 @@ impl InformationSchemaProvider {
             catalog_manager,
         }
     }
-}
 
-impl InformationSchemaProvider {
-    pub fn table(&self, name: &str) -> Result<Option<TableRef>> {
-        let stream_builder = match name.to_ascii_lowercase().as_ref() {
-            TABLES => Arc::new(InformationSchemaTables::new(
-                self.catalog_name.clone(),
-                self.catalog_manager.clone(),
+    pub fn build_information_schema(
```

Comment:
Not sure but
```suggestion
    pub fn tables(
```

Reply:
`tables` makes sense in terms of its return value, comparing to the `table` which returns a single table. However input of these two functions varies, which indicates they operate at different levels.

I would suggest to use `build` as name.